src/druidry/data_source.py

```diff
@@ -311,6 +311,13 @@ def like_filter(f):
     return LikeFilter(
         dimension=f['left']['field'], pattern=pattern)
 
+def regex_filter(f):
```

Comment:
I think the name for this, can be easily interpreted as more generic than it is intended to be. Specifically, I think it's not intended to be a generic regex type filter, and is intended to be a "like filter" implemented via regex. So maybe adding the word "like" into the function name, would make that more clear i.e. `regex_like_filter`. What do you think?

Reply:
Will rename.